iroh-p2p/src/behaviour.rs

```diff
@@ -88,9 +88,14 @@ impl NodeBehaviour {
         let pub_key = local_key.public();
         let peer_id = pub_key.to_peer_id();
 
-        let bitswap = if config.bitswap {
+        let bitswap = if config.bitswap_client || config.bitswap_server {
             info!("init bitswap");
-            let bs_config = BitswapConfig::default();
+            // NOTE: server only mode is not implemented yet
+            let bs_config = if config.bitswap_server {
+                BitswapConfig::default()
+            } else {
+                BitswapConfig::default_client_mode()
```

Comment:
This is just a nitpick:

```suggestion
            let bs_config = config.bitswap_server
                .then(BitswapConfig::default)
                .or_else(ButswapConfig::default_client_mode);
```

Nice that rust is this expressive, but in this particular instance I am not fond of one way or another :laughing: just wanted to show off in this case, hope you don't mind :laughing:

Reply:
tried to implement as suggested.  Ran into an issue:
`then` and `or_else` return `Option<T>`, which requires unwrapping, making it less clear than the `if / else` variant.